remo/profiles/models.py

```diff
@@ -251,9 +252,9 @@ def __unicode__(self):
 
     def save(self, *args, **kwargs):
         # Save the timestamp when a Rep becomes available
-        if not self.id:
+        if not self.id and self.start_date <= get_date():
             self.is_unavailable = True
-        else:
+        elif self.start_date < get_date():
```

Comment:
I restructured the code here to be more clear. The poorly implemented idea here is, that when calling the `save` method, the status needs to change under certain circumstances. For example you don't need to update the `is_available` flag if the start date is in the future, simply because the user in not in an unavailable status yet, but rather has expressed an interest of being unavailable in the future.